### PR TITLE
Add initial windows packer templates for ami

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Add initial windows packer templates for ami

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/2306

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: packer/scripts/windows/pip-install.ps1 [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+# This needs to be repeated more than twice (sometimes) to actually install packages without --user
+# After applying the install-pep-514.reg from scoop for Python specifically
+
+# Need TLS12 in order to install pip correctly
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+wget https://bootstrap.pypa.io/get-pip.py -OutFile get-pip.py
```
Comment (prudhvigodithi) [inline packer/scripts/windows/pip-install.ps1:12]: @peterzhuamazon is wget installed by default ? Consider using `Invoke-WebRequest` as this default windows powershell.
`Invoke-WebRequest -Uri "https://bootstrap.pypa.io/get-pip.py " -OutFile "get-pip.py"`

File: packer/scripts/windows/scoop-install-scoop.ps1 [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (prudhvigodithi) [inline packer/scripts/windows/scoop-install-scoop.ps1:1]: @peterzhuamazon can we rename this file to scoop-config?

File: packer/jenkins-agent-win2016-x64.json [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      "ami_name":"{{user `name-base`}}-{{user `build-time`}}",
+      "vpc_id":"{{user `build-vpc`}}",
+      "subnet_id":"{{user `build-subnet`}}",
+      "security_group_ids":"{{user `build-secgrp`}}",
+      "source_ami_filter":{
+        "filters":{
+          "virtualization-type":"hvm",
+          "architecture":"x86_64",
+          "name": "Windows_Server-2016-English-Full-Base-20*",
+          "block-device-mapping.volume-type":"gp2",
```
Comment (prudhvigodithi) [inline packer/jenkins-agent-win2016-x64.json:29]: Using `gp3` is better.

File: packer/scripts/windows/pip-install.ps1 (hunk 1 above)
Comment (peterzhuamazon) [reply packer/scripts/windows/pip-install.ps1:12]: `wget` in powershell is a default alias to `Invoke-WebRequest`. They are same.

File: packer/jenkins-agent-win2016-x64.json (hunk 3 above)
Comment (peterzhuamazon) [reply packer/jenkins-agent-win2016-x64.json:29]: gp3 has lag / latency in practice tho I would put this to gp2 for now.
Pretty easy to rebuild right now with packer.
gp2 can ebs optimize to 900 IOPS which is not bad during jenkins test.

File: packer/scripts/windows/scoop-install-scoop.ps1 (hunk 2 above)
Comment (peterzhuamazon) [reply packer/scripts/windows/scoop-install-scoop.ps1:1]: Can rename later.

File: packer/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# Needs to be run in this directory
+$ cd packer/
+
+# Run build AMI:
+$ packer build <template json name>
+
+# Run build AMI with debug mode:
+$ packer build -debug <template json name>
+```
+
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply packer/README.md:22]: Added now.

File: packer/scripts/windows/scoop-install-scoop.ps1 (hunk 2 above)
Comment (peterzhuamazon) [reply packer/scripts/windows/scoop-install-scoop.ps1]: Renamed now.

**Review by prudhvigodithi — APPROVED** [review]

**Pull request merged**